Updated project to handle Ruby 2.7.3

hawkpf opened the pull request:
Hello, I updated your project to handle ruby 2.7.3 to deploy on heroku's 20 stack.  Heroku-18 is no longer supported which was used to deploy Ruby 2.5 apps.

jamiefolsom commented:
Hi @hawkpf, thanks for this PR. Would you rebase it against the current master branch? At that point, we'll discuss with the project lead as to allocation of time to get it merged, tested, etc. Thanks! cc @blms @MartinFoys

blms commented:
Hi @hawkpf, I wanted to ask if you checked for breaking changes in any of the upgraded gems (or in Ruby itself) that might impact our project? If not, we might do a more thorough investigation before upgrading, in which case you don't need to worry about additional work (such as rebasing) on your PR.

hawkpf commented:
When I started to deploy DM2.0 to heroku I ran in to a wall because the current repo version or ruby is not supported by heroku's stack-20.  I updated the version of ruby to 2.7.3 then did a gem update.  There were still failures on bringing up the site so I used the debug info which pointed to mimemagic (0.3.3) being a problem.  I found that mimemagic (0.3.3) was replaced by marcel (1.0.1). https://www.theregister.com/2021/03/29/rails_mime_fix/ so I removed mimemagic and updated marcel to 1.0.1.  After that modification, the gems were squared away.  Next when I tried to launch the application from heroku it wasn't loading the index.html file specified by this error ActionController::RoutingError (No route matches [GET] "/").  That's why I copied the index.html file from the client/public/index.html directory to the public/ directory

blms commented:
Thanks for the work to sync changes with the current `master`, @hawkpf. However, I fear you may be duplicating the work we have done in https://github.com/performant-software/dm-2/pull/390! It's not yet merged, and it will only live on the `develop` branch for now, but it should cover the efforts you've made here.
